draft-ietf-tls-esni.md

```diff
@@ -540,8 +546,11 @@ it does a standard ClientHello, with the exception of the following rules:
    {{flow-client-reaction}}.)
 1. It MUST include an "encrypted_client_hello" extension with a payload
    constructed as described below.
-1. The value of `ECHConfig.contents.public_name` MUST be placed in the
-   "server_name" extension.
+1. The "server_name" extension MUST be constructed based on the reference
+   identity in `ECHConfig.contents.public_identity`. Specifically, if
+   `ECHConfig.contents.public_identity` is a DNS name then this name is
+   set in the "server_name" extension. Otherwise, the "server_name" extension
+   is omitted.
```

Comment:
A name like `0x42.0x42.0x42.0x42` would pass the DNS name validation, yet still be treated as an IP address by application code. I think it's worth acknowledging this possibility, so that clients cannot be induced into sending noncompliant server_name extensions to unsuspecting servers.

Reply:
> yet still be treated as an IP address by application code.

It would? 🤔

Reply:
Haha yes, this is an odd anti-feature, but I can verify it exists in Chrome and Safari. Visit http://0xac.0xd9.0xa.0xe, which is translated to google.com's IPv4 address: 172.217.10.14.

Reply:
(To clarify, I just think it's plausible that application code could get confused by something like this. The above example shows lax client-side interpretation, not any server-side confusion.)

Reply:
Wow, interesting. Thanks for clarifying!